internal/api/mfa.go

```diff
@@ -224,13 +224,13 @@ func (a *API) VerifyFactor(w http.ResponseWriter, r *http.Request) error {
 	challenge, err := models.FindChallengeByChallengeID(a.db, params.ChallengeID)
 	if err != nil {
 		if models.IsNotFoundError(err) {
-			return notFoundError(err.Error())
+			return notFoundError(ErrorCodeMFAFactorNotFound, "MFA factor with the provided challenge ID not found")
 		}
 		return internalServerError("Database error finding Challenge").WithInternalError(err)
 	}
 
 	if challenge.VerifiedAt != nil || challenge.IPAddress != currentIP {
-		return badRequestError("Challenge and verify IP addresses mismatch")
+		return unprocessableEntityError(ErrorCodeMFAIPAddressMismatch, "Challenge and verify IP addresses mismatch")
```

Comment:
Potential legitimate HTTP status code change, though unlikely to be handled by developers.

Reply:
Should be OK.